test-driven-range.js

```diff
@@ -0,0 +1,71 @@
+function isNotEmpty(data) {
+  if (data.length > 0) {
+    return true;
+  }
+  return false;
+}
+
+function isArray(data) {
+  if (Array.isArray(data)) {
+    return true;
+  }
+  return false;
+}
+
+function formRange(data) {
+  if (!isNotEmpty(data) || !isArray(data)) {
+    return 'Wrong Input';
+  }
+
+  let sortedInput = Object.keys(data);
+  let startAt = 0;
+  let endsAt;
+  let occurence = 1;
+  let ranges = [];
+
+  sortedInput.forEach((element, index) => {
+    const difference = getDifference(
+      sortedInput[index + 1],

```

Comment:
Accessing index+1 element is an undefined behavior for the last element. Please confirm that this will not pose any problem later if the JS behavior changes. 
In other words design should make sure that access to out of bounds elements is not performed. This will ensure consistency.